core/src/services/azdls/core.rs

```diff
@@ -97,17 +96,17 @@ impl AzdlsCore {
 }
 
 impl AzdlsCore {
-    pub async fn azdls_read(&self, path: &str, range: BytesRange) -> Result<Response<HttpBody>> {
-        let p = build_abs_path(&self.root, path);
-
-        let url = format!(
+    fn build_path_url(&self, path: &str) -> String {
+        format!(
             "{}/{}/{}",
             self.endpoint,
             self.filesystem,
-            percent_encode_path(&p)
-        );
+            percent_encode_path(build_abs_path(&self.root, path).trim_end_matches('/'))
```

Comment:
Not every path needs to be handled this way. I prefer to repeat some code instead to make it clear.

Reply:
fixed